### PR TITLE
Improve/Implement CRT system() and wsystem() 

### PR DESCRIPTION
**Pull request by encodedpr**
improve system() and wsystem() and helper function do_spawnT [WIP]

## Purpose

Simple fixes done to CRT regarding system() and wsystem() using wine code

JIRA issue: N/A

## TODO

Test runtime result, check existing wine tests or create more tests

### CODE REVIEW

File: sdk/lib/crt/process/_system.c [hunk 1]
```diff
@@ -14,110 +14,85 @@
...
 
-  if (command == NULL) {
-    if (szComSpec == NULL)
-      return 0;
-    else
-      return 1;
+  if (!(len = GetEnvironmentVariableW(comspec, NULL, 0))) len = sizeof(cmd)/sizeof(wchar_t);
+  if ((ret = HeapAlloc(GetProcessHeap(), 0, len * sizeof(wchar_t))))
+  {
+    if (!GetEnvironmentVariableW(comspec, ret, len)) strcpyW(ret, cmd);
```
Comment (HBelusca) [inline sdk/lib/crt/process/_system.c]: cmd without extension?

File: sdk/lib/crt/process/process.c [hunk 2]
```diff
@@ -309,23 +311,25 @@ do_spawnT(int mode, const _TCHAR* cmdname, const _TCHAR* args, const _TCHAR* env
...
-         return( 0);
+         CloseHandle(ProcessInformation.hProcess);	
+		 ProcessInformation.hProcess = 0;
+		/* fall through */		 
+      case _P_NOWAIT:
+      case _P_NOWAITO:
+         CloseHandle(ProcessInformation.hThread);
+         return((intptr_t)ProcessInformation.hProcess);
+      case _P_OVERLAY:
+         _exit(0);
```
Comment (HBelusca) [inline sdk/lib/crt/process/process.c]: leaking both hThread and hProcess?

**Comment by HBelusca** [comment]
where are the actual improvements?

File: sdk/lib/crt/process/_system.c (hunk 1 above)
Comment (encodedpr) [reply sdk/lib/crt/process/_system.c]: ComSpec=C:\WINDOWS\system32\cmd.exe

Comment (HBelusca) [reply sdk/lib/crt/process/_system.c]: I meant, the fall back string "cmd" (that you copy using strcpyW) does not contain any extension.

Comment (encodedpr) [reply sdk/lib/crt/process/_system.c]: addressed

File: sdk/lib/crt/process/_system.c [hunk 3]
```diff
@@ -14,110 +14,81 @@
...
-  if (szCmdLine == NULL)
+  len = strlenW(comspec) + strlenW(flag) + strlenW(cmd) + 1;
+  
+  if (!(fullcmd = HeapAlloc(GetProcessHeap(), 0, len * sizeof(wchar_t))))
   {
-     _set_errno(ENOMEM);
-     return -1;
+    HeapFree(GetProcessHeap(), 0, comspec);
+	_set_errno(ENOMEM);
+    return -1;
```
Comment (Fraizeraust) [inline sdk/lib/crt/process/_system.c]: Wrong indent on 63 line (unnecessary blank space). Care to fix it please?

**Comment by learn-more** [comment]
Test run: https://build.reactos.org/builders/Test%20KVM/builds/20628

**Pull request closed without merging**